crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -126,7 +139,22 @@ contract CommonBridge is
             gasLimit: 21000 * 5,
             data: bytes("")
         });
-        _deposit(depositValues);
+        _deposit(msg.sender, depositValues);
+    }
+
+    function depositERC20(address tokenL1, address tokenL2, address destination, uint256 amount) external {
+        require(amount > 0, "CommonBridge: amount to deposit is zero");
+        IERC20(tokenL1).safeTransferFrom(msg.sender, address(this), amount);
+
+        depositsERC20[tokenL1][tokenL2] += amount;
```

Comment:
Should we take care of reentrancy here?

Reply:
Let's remove that by swapping these two lines, just to be sure.

I don't think any attacks are possible, though, since here we're not sending any value out of the bridge. Any token transfers that result in recursive deposits will only further increase this amount, checking that the transfer doesn't revert any of those times.